routemaster/tests/test_cron_processors.py

```diff
@@ -13,6 +14,19 @@
     MetadataTimezoneAwareProcessor,
 )
 
+UTC = dateutil.tz.gettz('UTC')
+
+
+def recently() -> datetime.datetime:
+    """
+    Helper for getting a time shortly before now. This is mostly expected to be
+    used for the construction of cron processors, so that their construction
+    time is separate to (and earlier than) the current time when they're run
+    but otherwise very close (to avoid changing the semantics of the test).
+    """
```

Comment:
Can we not handle this with less than or equal to vs greater than behaviour in filtering?

Reply:
I'm not sure what filtering that would be?

Essentially what this is doing is ensuring that the time range (between construction and call) which gets passed to `time_appears_in_range` is non-instantaneous. In the case where that range is instantaneous it always returns `False` because it does `start < instant <= end`.

We could change `time_appears_in_range` (to be `start <= instant <= end`), however my concern would then be that the cron processing might happen twice if the moment it ran was exactly on the requested trigger time.

Reply:
We chatted IRL.

Reply:
111a634 clarifies the handling.